### ISSUE TITLE
DynDNS not working after upgrade to 0.91

### ISSUE THREAD

**Issue by mray** [freedombox/Plinth#444]
After upgrading to 0.91 I experienced the seeming de-installation of *all* applications.
After first attempts to rectify things everything seemed to work fine without an issue except DynDNS.

I had to provide all form informations again; but this time without success. I can't reach my box from the outside anymore.

**Comment by mray** [comment]
Since my 0.91 installation does not work i put a backup version with Plinth version 0.8.2 on again.
"Funny" enough I had to install everything there again, too. I hope DynDNS will work at least here when re-configured :/

**Comment by mray** [comment]
Maybe my backup technique (creating img files of the main partition on the sd cards) is the culprit?

**Comment by SunilMohanAdapa** [comment]
@mray I have investigated more about the problem you are facing and found the following:

- The script is falsely reporting success even when it failed.  This is happening because the provider dyndns.strato.com is reporting success in HTTP response (200 OK).  But there is a custom error they are sending in the response body that a generic script can't understand.  Due to this, when a request is sent, it is recorded as success and not retried for 1 hour.  So, if you have set wrong parameters first time and then later corrected them, then the new parameters will not be reflected for 1 hour.  So, please try freshly again to check if our setup was working.  I think this is mostly the case and your problem should get resolved.
- The script may handler special characters in username and password properly.  If you are using special characters in password, please try simplifying upper/lower case alphabets and numbers should be okay

More debugging steps:
- To check if things are running fine, please run the commands `curl ipinfo.io` and `host freedombox.mydomain.de` and see if the IP addresses returned by them are the same.
- Remove the file /etc/ez-ipupdate/last-update to make the script think an update was never done.
- Then run the script on the command line `bash -x /usr/share/plinth/actions/dynamicdns update`. This will show a lot of output on the console.  Please post it after masking your username and password in it.
- Above command should also create a file like `update?hostname=freedombox.mydomain.de` (actually this is a bug) in your current directory.  Please post the (one line) content of latest such file (there may be multiple).
- If all else fails, please run the command `wget -4 -o /dev/null -O - --user myusername --password mypassword https://dyndns.strato.com/nic/update?hostname=freedombox.mydomain.de`.  Then check if that successfully updates your IP and post its output.

**Comment by mray** [comment]
As you mentioned there problem probably was the DynDNS app, not FBX core. I got in touch with the DynDNS author and he advised me to do this:
`/usr/share/plinth/actions/dynamicdns clean`

After that I needed to re-enter all information into plinth and.
I checked `curl ipinfo.io` and `host freedombox.mydomain.de` and the IPs finally matched!
Now my box is reachable again through strato.de DynDNS service.

Thank you for taking care of this, I would have continued to fail without that insight.
My personal itch has been scratched, but depending on what needs to be done one the code I'll leave it up to you to close the issue.
